connect/runtime/src/test/java/org/apache/kafka/connect/runtime/rest/resources/ConnectorsResourceTest.java

```diff
@@ -196,20 +210,62 @@ public void testCreateConnectorExists() throws Throwable {
         PowerMock.verifyAll();
     }
 
-    @Test(expected = BadRequestException.class)
-    public void testCreateConnectorWithASlashInItsName() throws Throwable {
-        String badConnectorName = CONNECTOR_NAME + "/" + "test";
+    @Test
+    public void testCreateConnectorNameTrimWhitespaces() throws Throwable {
+        // Clone CONNECTOR_CONFIG_WITHOUT_NAME Map, as createConnector changes it (puts the name in it) and this
```

Comment:
this might be a good reason to make it a `static` method to get an instance of it instead of making it a `static` field. it could be easy to misuse if left as a field.

this is a good catch. i'd suggest making a copy in `createConnector` but since it's the entry point for a REST call it shouldn't ever be necessary.

Reply:
You are right, I'll move the cloning code to a static method.